src/features/defi/components/Approve/Approve.tsx

```diff
@@ -18,11 +20,13 @@ type ApproveProps = {
   cryptoEstimatedGasFee: string
   disableAction?: boolean
   fiatEstimatedGasFee: string
+  isExactAllowance?: boolean
   learnMoreLink?: string
   loading: boolean
   loadingText?: string
   contractAddress: string
   preFooter?: React.ReactNode
+  onSwitch?(): void
```

Comment:
I would prefer the toggle terminology here (chakra-ui uses `onChange` but I don't think it makes a lot of sense for the `<Approve />` component that's not as atomic as a switch)